askama/src/filters/json.rs

```diff
@@ -19,17 +19,74 @@ use serde_json::to_writer_pretty;
 /// or in apostrophes with the (optional) safe filter `'{{data|json|safe}}'`.
 /// In HTML texts the output of e.g. `<pre>{{data|json|safe}}</pre>` is safe, too.
 #[inline]
-pub fn json<S: Serialize>(s: S) -> Result<impl fmt::Display, Infallible> {
-    Ok(ToJson(s))
+pub fn json<S: Serialize, I: AsIndent>(
+    value: S,
+    indent: I,
+) -> Result<impl fmt::Display, Infallible> {
+    Ok(ToJson { value, indent })
 }
 
-#[derive(Debug, Clone)]
-struct ToJson<S: Serialize>(S);
+pub trait AsIndent {
+    fn as_indent(&self) -> Option<&str>;
+}
+
+impl AsIndent for str {
+    #[inline]
+    fn as_indent(&self) -> Option<&str> {
+        Some(self)
+    }
+}
+
+impl AsIndent for isize {
```

Comment:
Does this still work if we impl for `usize` directly?

Reply:
Not if it should support jinja's API where negative value = no beautification.